contracts/erc20-gateway/ERC20Cogateway.sol

```diff
@@ -88,4 +111,91 @@ contract ERC20Cogateway is
 
         emit GatewayProven(messageInbox, _blockNumber);
     }
+
+    /**
+     * @notice It allows withdrawing Utility tokens. Withdrawer needs to
+     *       approve erc20 cogateway contract for the amount to be
+     *       withdrawn.
+     *
+     * @param _utilityToken Address of utility token
+     * @param _amount Amount of tokens to be withdrawn.
+     * @param _beneficiary The address in the origin chain where the value
+     *                     where the tokens will be withdrawn.
+     * @param _feeGasPrice Gas price at which fee will be calculated.
+     * @param _feeGasLimit Gas limit at which fee will be capped.
+     *
+     * @return messageHash_ Message hash.
+     *
+     * \pre `_utilityToken` is not 0.
+     * \pre `_amount` is not 0.
+     * \pre `_beneficiary` is not 0.
+     * \pre `_amount` should be greater than maximum reward.
+     * \pre `msg.sender` should approve erc20 cogateway contract.
+     *
+     * \post Update the nonces storage mapping variable by incrementing the
+     *       value for `msg.sender` by one.
+     * \post Adds a new entry in `outbox` mapping storage variable. The value is
+     *       set as `true` for `messagehash_` in `outbox` mapping. The
+     *       `messageHash_` is calculated by `MessageOutbox.declareMessage`.
+     * \post Utility tokens will be burned from `msg.sender` account
+     *       by erc20 cogateway.
+     */
+    function withdraw(
+        address _utilityToken,
+        uint256 _amount,
+        address _beneficiary,
+        uint256 _feeGasPrice,
+        uint256 _feeGasLimit
+    )
+        external
+        returns(bytes32 messageHash_)
+    {
+        require(
+            _utilityToken != address(0),
+            "Utility Token address must not be 0."
+        );
+        require(
+            _amount != 0,
+            "Withdrawal amount should be greater than 0."
```

Comment:
```suggestion
            "Withdrawal amount should not be 0."
```

Maybe we don't need this check at all. The require at L166 is covering this check too.

Reply:
Yes, require at L166 is covering same check. Done.